src/tools/idlc/src/idlc.c

```diff
@@ -544,6 +546,15 @@ static const idlc_option_t *compopts[] = {
     "Enable or disable warnings. Possible values are: no-implicit-extensibility, "
     "no-extra-token-directive, no-unknown_escape_seq, no-inherit-appendable, "
     "no-eof-newline, no-enum-consecutive. " },
+  &(idlc_option_t){
+    IDLC_STRING, { .string = &config.output_dir }, 'd', "", "<directory>",
+    "Set the output directory for compiled files." },
+  &(idlc_option_t){
+    IDLC_STRING, { .string = &config.base_dir }, 'b', "", "<directory>",
+    "Enable directory reconstruction starting from <base_dir> "
+    "Attempts to recreate directory structure matching input. "
+    "Use without the -d option will compile IDL files in-place. "
+    "Fail-silent if root_dir is not a parent of input" },
```

Comment:
See above — there may be a perfectly valid reason for preferring this behaviour, perhaps is obvious as guarding against possible undetectable edge cases and TOCTTOU issues if someone is moving files at the same time. My inclination is that it would be best to consider it an error.

Reply:
I guess that makes sense. My thought process was that it shouldn't be allowed to create directories backwards from the "base folder", since that defeats the purpose of having preserved directory structure. They should be in the same tree.
But if that fails they could still compile in the working/output dir if they fail that check? I guess writing it out now it makes more sense just to not allow it.
Do you agree that it should fail with a potential reverse path though?